### PR TITLE
chore: upgrade to next 14

### PR DESCRIPTION
**Pull request by bautistaaa**

### CODE REVIEW

File: packages/ui/src/components/badge.tsx [hunk 1]
```diff
@@ -1,5 +1,3 @@
-'use client';
```
Comment (bautistaaa) [inline packages/ui/src/components/badge.tsx:1]: i dont understand why it hates this

File: packages/ui/src/components/card.tsx [hunk 2]
```diff
@@ -1,4 +1,3 @@
-'use client';
```
Comment (bautistaaa) [inline packages/ui/src/components/card.tsx:1]: i dont understand why it hates this

File: apps/web/src/app/[locale]/tracks/_components/track.action.ts [hunk 3]
```diff
@@ -109,3 +109,43 @@ export const getTrackDetails = cache(async (slug: string) => {
     },
   });
 });
+
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/tracks/_components/track.action.ts:112]: moved these here to see as an attempt to fix the enrolled track session errors

File: apps/web/src/app/[locale]/tracks/_components/track-enrolled-section.tsx [hunk 4]
```diff
@@ -52,17 +32,7 @@ export async function EnrolledTrackSection() {
...
-                <Link
-                  className="group w-[250px] flex-none snap-center focus:outline-none sm:w-[330px] xl:w-[333px]"
-                  href={`/tracks/${t.slug}`}
-                  key={t.id}
-                >
-                  <PersonalTrackCard track={t} />
-                </Link>
-              ))}
-            </Carousel>
+            <EnrolledTrackSectionCarousel tracks={tracks} />
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/tracks/_components/track-enrolled-section.tsx:35]: this is the last problem child.

attempted to pass the track props to this component with no luck

File: apps/web/src/app/[locale]/tracks/_components/track-enrolled-section-carousel.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import Link from 'next/link';
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/tracks/_components/track-enrolled-section-carousel.tsx:1]: broke this out to see if it fixed anything

File: apps/web/src/app/[locale]/tracks/_components/track-enrolled-section-carousel.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    <>
+      <div className="container p-4">
+        <h2 className="relative text-3xl font-bold tracking-tight">
+          <div className="absolute -left-8 -z-10 h-12 w-32 rounded-full bg-blue-300/50 blur-3xl" />
+          Your Tracks
+        </h2>
+      </div>
+
+      <section className="relative flex w-full flex-row gap-4 overflow-hidden rounded-[2.5rem]">
+        <Carousel>
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/tracks/_components/track-enrolled-section-carousel.tsx:17]: no matter what `Carousel` breaks with the cyclic error. Even if I just have `Carousel` return 
'hi'

File: apps/web/src/app/[locale]/tracks/_components/SkeletonTrack.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Card } from '@repo/ui/components/card';
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/tracks/_components/SkeletonTrack.tsx:1]: broke this out in hopes to fix Enrolled Section errros

File: apps/web/src/app/[locale]/tracks/_components/track-enrolled-section-carousel.tsx (hunk 6 above)
Comment (smakosh) [reply apps/web/src/app/[locale]/tracks/_components/track-enrolled-section-carousel.tsx:17]: Go with `<Carousel tracks={tracks} />` and move everything in there.

**Comment by ctjlewis** [comment]
lgtm 🫡

**Comment by bautistaaa** [comment]
thanks so much @ctjlewis

**Pull request merged**